yarn-project/end-to-end/src/e2e_fees/private_refunds.test.ts

```diff
@@ -86,8 +83,6 @@ describe('e2e_fees/private_refunds', () => {
     // the fee limit minus the final transaction fee. The other 2 fields in the note are Alice's npk_m_hash and
     // the randomness.
     const refundNoteValue = t.gasSettings.getFeeLimit().sub(new Fr(tx.transactionFee!));
-    // TODO(#7324): The values in complete address are currently not updated after the keys are rotated so this does
-    // not work with key rotation as the key might be the old one and then we would fetch a new one in the contract.
```

Comment:
The comment I removed here still applies but it's not related to the issue #7324 but instead to unfinished key rotation.

@nventuro given that we've decided to keep key rotation (right?) we should create an issue for this. What we would want here is to fetch npk_m_hash which was valid in the given tx (I assume at the start of the tx). It seems reasonable that we would add an endpoint to AztecNode for this. It could look something like:
`fetchNpkMHash(txHash: TxHash): Promise<Fr>`

This will most likely be quite complicated to do because we currently don't have the data needed in aztec node. We will probably need to build a database tracking key rotation changes.

Reply:
That is unfortunately not enough - under key rotation, the sender would not necessarily be forced to use the most up-to-date npk: they are free to use a slightly old one. So you need to know which one they used in order to convince yourself that the note is real.

Becuse we don't want to increase DA costs by including the npk, the idea is for PXE to try with the different npks that were valid at that block until it gets a matching note hash. As you point out, the key registry would emit events that'd let you track this information. I'd imagine that `pxe.addNote` would contain this behavior once we add it.